File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s) != len(t)):
            return False
        
        d = defaultdict(lambda: 0)

        for ch in s:
            d[ch] += 1
        
        for ch in t:
            d[ch] -= 1

        for val in d.values():
            if val:
                return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return false;

        unordered_map<char, int> mp;

        for(int i = 0; i < s.size(); i++){
            mp[s[i]]++;
        }

        for(int i = 0; i < t.size(); i++){
            if(mp.find(t[i]) != mp.end()){
                mp[t[i]]--;
                if(!mp[t[i]]){
                    mp.erase(t[i]);
                }
            } else {
                return false;
            }
        }

        return mp.size() ? false : true;

        return true;
    }
};